### PR TITLE
Rename and deprecate underscored methods for J4

### PR DESCRIPTION
**Pull request by wilsonge**
in JFilter

---
### How to test
- apply this patch
- navigate to the backend and frontend and be sure nothing is broken
- report your result back here ;)

### CODE REVIEW

File: libraries/joomla/filter/input.php [hunk 1]
```diff
@@ -647,10 +647,25 @@ protected static function decodeFileData(array $data)
 	 *
 	 * @return  string  'Cleaned' version of input parameter
 	 *
-	 * @since   11.1
+	 * @since      11.1
+	 * @deprecated 3.5   This will be removed in 4.0 in favour of the non-prefixed method
```
Comment (mbabker) [inline libraries/joomla/filter/input.php]: `@deprecated  4.0  Use JFilterInput::remove() instead`

**Comment by zero-24** [comment]
@wilsonge how can this be tested? Just "joomla still works"?

**Comment by wilsonge** [comment]
Yeah. Filter is used literally on every page tens/hundreds of times. So it *should* be pretty obvious if it is broken

**Comment by zero-24** [comment]
Thanks :smile: 

### How to test

- apply this patch
- navigate to the backend and frontend and be sure nothing is broken
- report your result back here ;)

**Comment by alikon** [comment]
nothing seems to be broken to me

**Comment by KingLouis1** [comment]
I have tested this item :white_check_mark: successfully on 30cd89edee24587fd3173a50819403e9a8b7e059<br /><br />No broken components etc. found in front and backend<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8047">issues.joomla.org/joomla-cms/8047</a>.</sub>

**Comment by Bakual** [comment]
Setting RTC based on tests and review.

**Comment by Kubik-Rubik** [comment]
Thank you @wilsonge and testers! Merged.

**Pull request merged**